Tasks/CopyFiles/task.json

```diff
@@ -0,0 +1,87 @@
+﻿{
+  "id": "5BFB729A-A7C8-4A78-A7C3-8D717BB7C13C",
+  "name": "CopyFiles",
+  "friendlyName": "Copy Files",
+  "description": "Copy files from source folder to target folder using minimatch patterns",
+  "helpMarkDown": "[More Information](http://go.microsoft.com/fwlink/?LinkID=)",
+  "category": "Utility",
+  "visibility": [
+    "Build"
+  ],
+  "author": "Microsoft Corporation",
+  "version": {
+    "Major": 1,
+    "Minor": 0,
+    "Patch": 0
+  },
+  "demands": [
+  ],
+  "minimumAgentVersion": "1.91.0",
+  "groups": [
+        {
+            "name": "advanced",
+            "displayName": "Advanced",
+            "isExpanded": false
+        }
+    ],
+  "inputs": [
+    {
+      "name": "SourceFolder",
+      "type": "string",
+      "label": "Source Folder",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "Source folder to apply copy patterns to"
+    },  
+    {
+      "name": "Contents",
+      "type": "multiLine",
+      "label": "Contents",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "File paths to include as part of the copy. Supports multiple lines or minimatch patterns. [More Information](http://go.microsoft.com/fwlink/?LinkID=)"
+    },
+    {
+      "name": "TargetFolder",
+      "type": "string",
+      "label": "Target Folder",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "Target folder files will copy to"
+    },
+    {
+      "name": "CleanTargetFolder",
+      "type": "boolean",
+      "label": "Clean Target Folder",
+      "defaultValue": "false",
+      "required": false,
+      "helpMarkDown": "Delete all existing files in target folder before copy.",
+      "groupName":"advanced"
+    },
+    {
+      "name": "OverWrite",
+      "type": "boolean",
+      "label": "Over Write",
+      "defaultValue": "false",
+      "required": false,
+      "helpMarkDown": "Relace existing file in target folder",
+      "groupName":"advanced"
+    },
+    {
+      "name": "UseCommonRoot",
+      "type": "boolean",
+      "label": "Use Common Root",
+      "defaultValue": "false",
```

Comment:
I think what we are really trying to accomplish here is to flatten the folder hierarchy and maybe that should be the option.

I am not sure I like the idea of having the user specify the common root because I think it is hard to explain and specify.  We already have enough issues explaining the copy patterns in the existing task and how those apply to the Copy Root.

From: Bryan MacFarlane [mailto:notifications@github.com]
Sent: Tuesday, November 3, 2015 9:00 AM
To: Microsoft/vso-agent-tasks <vso-agent-tasks@noreply.github.com>
Cc: Chris Patterson <Chris.Patterson@microsoft.com>
Subject: Re: [vso-agent-tasks] Users/tihuang/newartifact (#698)


In Tasks/CopyFiles/task.json<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fMicrosoft%2fvso-agent-tasks%2fpull%2f698%23discussion_r43774907&data=01%7c01%7cChris.Patterson%40microsoft.com%7c8ddc81f5d2d640130b6808d2e4703f9a%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=xpPqIgh8ZnIGJl7kRpE%2fVsDRlzt%2baciPhKqy5XZymkk%3d>:

> +      "groupName":"advanced"

> +    },

> +    {

> +      "name": "OverWrite",

> +      "type": "boolean",

> +      "label": "Over Write",

> +      "defaultValue": "false",

> +      "required": false,

> +      "helpMarkDown": "Relace existing file in target folder",

> +      "groupName":"advanced"

> +    },

> +    {

> +      "name": "UseCommonRoot",

> +      "type": "boolean",

> +      "label": "Use Common Root",

> +      "defaultValue": "false",

No we shouldn't default to true.

IMHO, common root seems very fragile. Other things that react post copy suddenly break as other changes are made.

Is it better to have an advanced optional argument which is the common root? c:\a\b\c
After all, I might want c:\a\b and end up with
d:\t\c\test*.txt in the target

Seems more deterministic, less spooky, and as an optional advanced option, more control

Thoughts?

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fMicrosoft%2fvso-agent-tasks%2fpull%2f698%2ffiles%23r43774907&data=01%7c01%7cChris.Patterson%40microsoft.com%7c8ddc81f5d2d640130b6808d2e4703f9a%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=D7ndJMrKhrRT83o%2f20lNRLrT9ji3PSdFi18yw4X4mhc%3d>.